### ISSUE TITLE
Compiler messages no longer included in smoke report

### ISSUE THREAD

**Issue by bram-perl** [abeltje/Test-Smoke#54]
The compiler warning/error messages are no longer included in smoke reports.

A random report that still has them (because it's an older version of Test::Smoke): https://www.nntp.perl.org/group/perl.daily-build.reports/2022/08/msg283943.html : 
```
Compiler messages(gcc):
/usr/include/sys/cdefs_elf.h:67:20: error: missing binary operator before token "("
/usr/include/sys/cdefs_elf.h:73:20: error: missing binary operator before token "("
/usr/include/sys/cdefs_elf.h:79:20: error: missing binary operator before token "("
...
```

Some context:

* 'smokecurrent.log' contains the stdout/stderr output of the 'tssmokeperl.pl' command.
* a 'mktest.out' log file is also created (in the build dir), data is written to this via `$self->log(...)`
* when creating the report the 'smokecurrent.log' file is used to search for compiler error/warning messages (`Test::Smoke::Util::grepccmsg`)

There are two commits that are relevant:

* *commit be0b7a9438bdc9b25176dc8676641aff3bfc6413:*
  * Before this commit Test::Smoke::Smoker::_run used:

        my @output = qx( $command );

      stdout of the `make` command (among others) is captured, the stderr error is not captured and ends up in 'smokecurrent.log'

  * After this commit:

        my ( $out, $err, $res ) = capture { system $command };
        print STDERR $err if $err;

      stderr of `$command` still goes to 'smokecurrent.log' so all is good;

* *commit 69c29d3918894a7b4d8ce26cabe5dcad4fdeabd7:* (included in v1.75)
  * Before:

        my ( $out, $err, $res ) = capture { system $command };
        print STDERR $err if $err;

  * After:

        my ( $out, $err, $res ) = capture { system $command };
        $self->log($err) if $err;

  * stderr of `$command` is now logged via `$self->log` which ends up in 'mktest.out' and not in 'smokecurrent.log'.
`Test::Smoke::Util::grepccmsg` however only looks at 'smokecurrent.log'
  * => compiler messages no longer included in the report.

**Possible fixes:**

* re-use `print STDERR $err`: this doesn't do the right thing on Windows (at least not in my test).
  When doing that the stdout and stderr becomes mixed in 'smokecurrent.log' which is bad.
  (see [smokecurrent - print STDERR.log](https://github.com/abeltje/Test-Smoke/files/9332022/smokecurrent.-.print.STDERR.log))
* use `$self->ttylog`: this logs the message in mktest.out *and* prints it on stdout with a timestamp (and stdout ends up in 'smokecurrent.log').
  This sort of works but causes the report to show the timestamp for the compiler messages which isn't that great.
* use `local $Test::Smoke::LogMixin:USE_TIMESTAMP=0` and `$self->ttylog`:
  This appears to work, no timestamp in the report and the compiler messages are included in the report

**Comment by abeltje** [comment]
First of all: *Thanks for this detailed analysis, really helpful with 20 years old code!*

Thinking about this, `Test::Smoke` shouldn't rely on the logfile as could be a terminal, I guess @Tux designed `mktest.out` to hold all the information to generate the report. With that in mind we have to make sure all the compiler messages end up in `mktest.out` and we use *that* as the source for `Test::Smoke::Util::grepccmsg()`.

I'll give that a go on my smokers and see what rolls out.

**Comment by abeltje** [comment]
The PR was merged the result will be in 1.80.
